python/ray/experimental/state/state_cli.py

```diff
@@ -277,35 +284,31 @@ def format_object_summary_output(state_data: Dict) -> str:
 
 
 def format_get_api_output(
-    state_data: Optional[Dict],
+    state_data: Optional[StateSchema],
     id: str,
     *,
     schema: StateSchema,
     format: AvailableFormat = AvailableFormat.YAML,
 ) -> str:
-
-    if not state_data or len(state_data) == 0:
+    if not state_data or isinstance(state_data, list) and len(state_data) == 0:
         return f"Resource with id={id} not found in the cluster."
-
-    return output_with_format(state_data, schema=schema, format=format)
+    if not isinstance(state_data, list):
+        state_data = [state_data]
+    state_data = [dataclasses.asdict(state) for state in state_data]
+    return output_with_format(state_data, schema=schema, format=format, detail=True)
 
 
 def format_list_api_output(
-    state_data: List[Dict],
+    state_data: List[StateSchema],
     *,
     schema: StateSchema,
     format: AvailableFormat = AvailableFormat.DEFAULT,
+    detail: bool = False,
 ) -> str:
     if len(state_data) == 0:
         return "No resource in the cluster"
-    if schema == TaskState and format == AvailableFormat.YAML:
```

Comment:
We don't need to do this anymore because with dataclass, task_id will always be the first field, so we don't have to extract it with YAML format.